app/playground/page.tsx

```diff
@@ -6,7 +6,7 @@ import {
 } from './_api';
 import { ICommodity, IPost } from '../_types';
 
-// Force dynamic fetching during runtime for playground
+// Avoid statically building this page during build.
```

Comment:
Could you explain the necessity of this please at the page.tsx level?

At this level I don't foresee us fetching anything that we don't want cached such as a list of commodity names that don't really change.

From what I understand NextJS will automatically [switch these pages to dynamically rendered](https://nextjs.org/docs/app/building-your-application/rendering/server-components#server-rendering-strategies) if uncached data is detected. It seems to go against the premise of caching if we were to force these routes to be dynamic.

Reply:
I had removed it in one of the commits in this PR based on the same understanding that it should not make these fetch calls during build but for some reason it was still making these calls during the github actions build. Have added this for now to fix that issue until I look into what's causing this to make these calls. However, it is needed in the other server-error scenario because I am not doing anything there for nextJs to detect that it should be done dynamically.

Reply:
I also found this useful implementation to simulate github actions when needed, more details on the purpose and intent - https://github.com/ed-pilots-network/frontend/issues/112. Will implement this to help with debugging these issues.

Reply:
> on the same understanding that it should not make these fetch calls during build

The calls that populate cached data at the page.tsx level should happen at build (atm we only do this to fetch an updated list of possible commodities). The data is required by the children. But you're probably talking specifically about the playground route, where you're right...though it's an exception to use 'no-cache' in our app.

From what I'm understanding here, there's a build error w/o this dynamic flag that occurs in the playground route because the server fetches are set to no-cache.

Reply:
Updated this to use cache config.

Reply:
It still fails - https://github.com/ed-pilots-network/frontend/actions/runs/6192011962/job/16811315676. I will debug this once I setup `act` for our package so I can do this locally. For now, have reverted back to forcing dynamic rendering.